x/mock/simulation/rand_util.go

```diff
@@ -37,14 +38,37 @@ func RandStringOfLength(r *rand.Rand, n int) string {
 }
 
 // Generate a random amount
+// Note: The range of RandomAmount includes max, and is, in fact, biased to return max.
 func RandomAmount(r *rand.Rand, max sdk.Int) sdk.Int {
-	return sdk.NewInt(int64(r.Intn(int(max.Int64()))))
+	// return sdk.NewInt(int64(r.Intn(int(max.Int64()))))
+	max2 := big.NewInt(0).Mul(max.BigInt(), big.NewInt(2))
+	randInt := big.NewInt(0).Rand(r, max2)
+	if randInt.Cmp(max.BigInt()) > 0 {
+		randInt = max.BigInt()
+	}
+	result := sdk.NewIntFromBigInt(randInt)
+	// Sanity
+	if result.GT(max) {
```

Comment:
result should be exclusive of max, so this is incorrect... will add pseudocode 1 min...

Reply:
I think that everywhere RandomAmount and RandomDecAmount are used, it makes sense to include the upper bound.  (Though, I agree that leaving out the upper bound is more common and intuitive.)